### ISSUE TITLE
--mangle incorrect output test case

### ISSUE THREAD

**Issue by knaccc** [terser/terser#1215]
**Bug report or Feature request?**

Bug

**Version (complete output of `terser -V` or specific git commit)**

terser 5.14.1

**Complete CLI command or `minify()` options used**

`terser --toplevel --mangle --mangle-props --output out.js in.js`

**`terser` input**

**input A:**

```js
function clearStorage() {
  chrome.storage.local.clear();
}
clearStorage();
```

**input B:**

```js
function clearStorage() {
  chrome.storage.local.clear();
}
clearStorage();
let obj = {chrome: 1, storage: 2, local: 3, clear: 4};
```

**`terser` output or error**

**output A:** (correct)

```js
function c(){chrome.storage.local.clear()}c();
```

**output B:** (incorrect)

```js
function c(){chrome.storage.l.clear()}c();let l={l:1};
```

**Expected result**

**output B:** should be:

```js
function c(){chrome.storage.local.clear()}c();let l={l:1};
```
or
```js
function c(){chrome.storage.local.clear()}c();let l={local:1};
```

**Comment by fabiosantoscode** [comment]
Looks like the browser extension API isn't in domprops.json! 

Is there an easy way to get a list of all the properties defined within, like what's done [here](https://github.com/terser/terser/blob/master/tools/props.html)? I haven't built a browser extension before.

**Comment by fabiosantoscode** [comment]
I agree with @evanw there BTW. It's better to opt-in to prop mangling (through a pattern) than to opt-out through quoting, as opting out can be error-prone (and break your dependencies if you have them).

But if your project is reasonably small I think it's more manageable.

**Comment by knaccc** [comment]
> Looks like the browser extension API isn't in domprops.json!
> 
> Is there an easy way to get a list of all the properties defined within, like what's done [here](https://github.com/terser/terser/blob/master/tools/props.html)? I haven't built a browser extension before.

The chrome.* APIs are not all present unless running within the context of an extension whose manifest.json file requests every possible API permission.

I discovered that IntelliJ IDEA uses [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/chrome/index.d.ts) to get type definitions for syntax highlighting. After I realized how much work it would be to parse it properly, I instead came up with this good-enough solution for my own purposes:

```bash
# get the type definitions
wget https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/master/types/chrome/index.d.ts

# remove all comments
cc -undef -E -P -xc index.d.ts > index.d.ts.2

# get every unique word that we may need to avoid mangling (needs GNU grep, not Mac grep)
grep -oP '\w+' index.d.ts.2 | grep -v '^[0-9]' | sort | uniq | less
```

**Comment by knaccc** [comment]
>  It's better to opt-in to prop mangling (through a pattern) than to opt-out through quoting, as opting out can be error-prone (and break your dependencies if you have them).
> But if your project is reasonably small I think it's more manageable.

I've used mangling on Javascript projects with tens of thousands of lines of code with terser, and it works very well. There are generally only a small number of places where things need to be quoted, and bugs there are very quickly discovered. In contrast, there would need to be a hundred thousand underscores introduced into the code in order to explicitly opt in to mangling.

**Comment by knaccc** [comment]
Btw referring back to the bug report at the beginning of the thread, I wanted to draw attention to the peculiarity that `local` is at first not mangled (even though it's not in the reserved list), but then suddenly mangled if it was used as a property elsewhere.

I was wondering why that happened, and if that inconsistent behaviour is a bug.
